gitpod-io/gitpod
# Fix supervisor panic when tkns and publicapi Service is nil

## Description


Review only commit https://github.com/gitpod-io/gitpod/pull/15274/commits/e081051ad939278d68faa63642fd1a8c38c362fe

Test also

```golang
package main

import (
	"fmt"
)

var t TestInterface
var t2 TestInterface

func main() {
	t = NewImplement()
	t.Echo("hello")

	t2 = NewImplement2()
	t2.Echo("hello")
}

type TestInterface interface {
	Echo(something string)
}

type Implement struct {
	hello *string
}

func NewImplement() *Implement {
	return nil
}

func NewImplement2() TestInterface {
	return nil
}

var _ TestInterface = (*Implement)(nil)

// Echo implements TestInterface
func (s *Implement) Echo(something string) {
	if s == nil {
		fmt.Println("s is nil")
		return
	}
	fmt.Println(*s.hello, something)
}

```

## Related Issue(s)

Fixes #

## How to test

- Open workspace with repo that will buildimage i.e. https://github.com/gitpod-io/openvscode-server
- Image build should success
- Workpsaces open should works

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
